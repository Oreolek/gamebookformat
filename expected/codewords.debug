BEGIN DEBUG OUTPUT
Book title: Gamebook
Number of sections: 9


Turn to 1 to begin.

1 (start) -  Demonstrating how [COLLECT code]Codewords[/COLLECT] (AKA sightings) can be used. Go to 2. 
2 (second) -  Got codeword [ADD code]warrior[/ADD]. Simple enough to set a codeword. Turn to 5. 
3 (xortest) -  OK, if you have the codeword [HAS code]warrior[/HAS], you may turn to 4 otherwise you may go back to 2. Although we both know you have that codeword. If you have the codeword [HAS code]fun[/HAS] you may turn to 6, without it you can go to 8. 
4 (the_end) -  That was easy.    
5 (choice) -  If you have the codeword [HAS code]warrior[/HAS] you may turn to 4. If you do not have the codeword [HAS NOT code]fun[/HAS NOT], you may turn to 8. If you have the codeword [HAS code]fun[/HAS], you may instead turn to 6. Otherwise see 3. 
6 (cheater) -  Cheater! There is no way you can have codeword fun. 
7 (noautotest) -  If you have the codeword [HAS code]fun[/HAS] turn to ((automatic))6. Otherwise you can go to the end at 4 or to the xortest at 3. 
8 (notsofun) -  This is just to demonstrate choices allowed when not having a codeword. Now go on to 9 (autotest) or 3 (xor test). 
9 (autotest) -  If you have codeword [HAS code]warrior[/HAS] turn to ((automatic))7. Otherwise you can go to the end at 4 or back to 1. 
END DEBUG OUTPUT
