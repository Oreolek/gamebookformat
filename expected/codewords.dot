digraph gamebook {

 1->2
2->3
3->4

3->9

3->5

3->6
4->8

4->6
6->4

6->5

6->2

6->9
7->5

7->6

7->9
8->9

8->7

8->1
}
