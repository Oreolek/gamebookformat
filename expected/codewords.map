{"notsofun": 8, "autotest": 7, "the_end": 3, "second": 2, "choice": 9, "xortest": 4, "start": 1, "noautotest": 6, "cheater": 5}