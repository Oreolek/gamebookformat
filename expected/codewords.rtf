{\rtf1\ansi\ansicpg1252\cocoartf1038\cocoasubrtf360
{\fonttbl\f0\fswiss\fcharset0 Helvetica;}
{\colortbl;\red255\green255\blue255;}
\paperw11900\paperh16840\margl1440\margr1440\vieww14140\viewh14860\viewkind0
\pard\tx566\tx1133\tx1700\tx2267\tx2834\tx3401\tx3968\tx4535\tx5102\tx5669\tx6236\tx6803\ql\qnatural\pardirnatural

\f0\b\fs24 \cf0

\b \qc Gamebook
\b0\

\b Turn to 1 to begin.
\b0\
\

\b \qc 1
\b0\
\ql  Demonstrating how Codewords (AKA sightings) can be used. Go to \b 2
\b0
.  \
\

\b \qc 2
\b0\
\ql  Got codeword warrior. Simple enough to set a codeword. Turn to \b 3
\b0
.  \
\

\b \qc 3
\b0\
\ql  If you have the codeword warrior you may turn to \b 9
\b0
. If you do not have the codeword fun, you may turn to \b 4
\b0
. If you have the codeword fun, you may instead turn to \b 5
\b0
. Otherwise see \b 6
\b0
.  \
\

\b \qc 4
\b0\
\ql  This is just to demonstrate choices allowed when not having a codeword. Now go on to \b 8
\b0
 (autotest) or \b 6
\b0
 (xor test).  \
\

\b \qc 5
\b0\
\ql  Cheater! There is no way you can have codeword fun.  \
\

\b \qc 6
\b0\
\ql  OK, if you have the codeword warrior, you may turn to \b 9
\b0
 otherwise you may go back to \b 2
\b0
. Although we both know you have that codeword. If you have the codeword fun you may turn to \b 5
\b0
, without it you can go to \b 4
\b0
.  \
\

\b \qc 7
\b0\
\ql  If you have the codeword fun turn to \b 5
\b0
. Otherwise you can go to the end at \b 9
\b0
 or to the xortest at \b 6
\b0
.  \
\

\b \qc 8
\b0\
\ql  If you have codeword warrior turn to \b 7
\b0
. Otherwise you can go to the end at \b 9
\b0
 or back to \b 1
\b0
.  \
\

\b \qc 9
\b0\
\ql  That was easy.     \
\

}
