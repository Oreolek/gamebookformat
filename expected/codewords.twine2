<tw-storydata name="Gamebook" startnode="1" creator="Twine" creator-version="2.0" format="Harlowe" options="">
<style role="stylesheet" id="twine-user-stylesheet" type="text/twine-css"></style><script role="script" id="twine-user-script" type="text/twine-javascript"></script>
<tw-passagedata pid="1" name="1" tags="" position="10,10"> Demonstrating how *Codewords* (AKA sightings) can be used. Go to [[2]]
. </tw-passagedata>
<tw-passagedata pid="2" name="2" tags="" position="20,20"> Got codeword *warrior*. Simple enough to set a codeword. Turn to [[9]]
. </tw-passagedata>
<tw-passagedata pid="3" name="3" tags="" position="30,30"> That was easy.    </tw-passagedata>
<tw-passagedata pid="4" name="4" tags="" position="40,40"> OK, if you have the codeword *warrior*, you may turn to [[3]]
 otherwise you may go back to [[2]]
. Although we both know you have that codeword. If you have the codeword *fun* you may turn to [[5]]
, without it you can go to [[8]]
. </tw-passagedata>
<tw-passagedata pid="5" name="5" tags="" position="50,50"> Cheater! There is no way you can have codeword fun. </tw-passagedata>
<tw-passagedata pid="6" name="6" tags="" position="60,60"> If you have the codeword *fun* turn to [[5]]
. Otherwise you can go to the end at [[3]]
 or to the xortest at [[4]]
. </tw-passagedata>
<tw-passagedata pid="7" name="7" tags="" position="70,70"> If you have codeword *warrior* turn to [[6]]
. Otherwise you can go to the end at [[3]]
 or back to [[1]]
. </tw-passagedata>
<tw-passagedata pid="8" name="8" tags="" position="80,80"> This is just to demonstrate choices allowed when not having a codeword. Now go on to [[7]]
 (autotest) or [[4]]
 (xor test). </tw-passagedata>
<tw-passagedata pid="9" name="9" tags="" position="90,90"> If you have the codeword *warrior* you may turn to [[3]]
. If you do not have the codeword *fun*, you may turn to [[8]]
. If you have the codeword *fun*, you may instead turn to [[5]]
. Otherwise see [[4]]
. </tw-passagedata>
</tw-storydata>
