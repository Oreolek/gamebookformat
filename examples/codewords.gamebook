* 1 start
Demonstrating how codewords (AKA sightings) can be used.