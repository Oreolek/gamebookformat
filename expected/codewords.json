{
  "title" : "Gamebook",
  "nr_sections" : "9",

  "sections" : {"IGNORE-debug-json-padding-IGNORE" : "", "start" : {
  "nr" : 1,
  "text" : ["" ," Demonstrating how ",{"collection" : "code", "name" : "Codewords"}
," (AKA sightings) can be used. Go to ",{"reference" : "2"},". "]
}
, "second" : {
  "nr" : 2,
  "text" : ["" ," Got codeword ",{"counter" : "code", "amount" : "warrior", "action" : "add"},". Simple enough to set a codeword. Turn to ",{"reference" : "9"},". "]
}
, "the_end" : {
  "nr" : 3,
  "text" : ["" ," That was easy.    "]
}
, "xortest" : {
  "nr" : 4,
  "text" : ["" ," OK, if you have the codeword ",{"collection" : "code", "what" : "warrior", "action" : "has"},", you may turn to ",{"modifier" : "xor"},{"reference" : "3"}," otherwise you may go back to ",{"reference" : "2"},". Although we both know you have that codeword. If you have the codeword ",{"collection" : "code", "what" : "fun", "action" : "has"}," you may turn to ",{"modifier" : "xor"},{"reference" : "5"},", without it you can go to ",{"reference" : "8"},". "]
}
, "cheater" : {
  "nr" : 5,
  "text" : ["" ," Cheater! There is no way you can have codeword fun. "]
}
, "noautotest" : {
  "nr" : 6,
  "text" : ["" ," If you have the codeword ",{"collection" : "code", "what" : "fun", "action" : "has"}," turn to ",{"modifier" : "auto"},{"reference" : "5"},". Otherwise you can go to the end at ",{"reference" : "3"}," or to the xortest at ",{"reference" : "4"},". "]
}
, "autotest" : {
  "nr" : 7,
  "text" : ["" ," If you have codeword ",{"collection" : "code", "what" : "warrior", "action" : "has"}," turn to ",{"modifier" : "auto"},{"reference" : "6"},". Otherwise you can go to the end at ",{"reference" : "3"}," or back to ",{"reference" : "1"},". "]
}
, "notsofun" : {
  "nr" : 8,
  "text" : ["" ," This is just to demonstrate choices allowed when not having a codeword. Now go on to ",{"reference" : "7"}," (autotest) or ",{"reference" : "4"}," (xor test). "]
}
, "choice" : {
  "nr" : 9,
  "text" : ["" ," If you have the codeword ",{"collection" : "code", "what" : "warrior", "action" : "has"}," you may turn to ",{"reference" : "3"},". If you do not have the codeword ",{"collection" : "code", "what" : "fun", "action" : "hasnot"},", you may turn to ",{"reference" : "8"},". If you have the codeword ",{"collection" : "code", "what" : "fun", "action" : "has"},", you may instead turn to ",{"reference" : "5"},". Otherwise see ",{"reference" : "4"},". "]
}
}}
